prog/dftb+/lib_dftbplus/parser.F90

```diff
@@ -237,25 +233,40 @@ subroutine readParserOptions(node, root, flags)
     type(fnode), pointer :: root
 
     !> Contains parser flags on exit.
-    type(TParserFlags), intent(out) :: flags
+    type(TParserFlags), intent(inout) :: flags
 
-    integer :: inputVersion
     type(fnode), pointer :: child
+    real(dp) :: rTmp, rVersion
+
+    ! Parser input version in form x.y with year and release number, matching code (after version 8)
+    call getChild(node, "ParserVersion", child, requested=.false.)
+    if (associated(child)) then
+      call getChildValue(child, "", rTmp, rVersion)
+      flags%version%major = nint(rTmp)
+      flags%version%minor = nint(10.0_dp*(rTmp - flags%version%major))
+      if (abs(rTmp - flags%version%major + 0.1_dp*flags%version%minor) > epsilon(1.0)) then
```

Comment:
This arithmetic error allows only integer values to pass the test (`20.0` will pass, `20.1` will always fail)
```suggestion
      if (abs(rTmp - flags%version%major - 0.1_dp*flags%version%minor) > epsilon(1.0)) then
```
I would still prefer to replace the entire thing.